platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/offline/OfflineManager.java

```diff
@@ -183,6 +205,53 @@ public void run() {
     });
   }
 
+  /**
+   * Merge offline regions from a secondary database into the main offline database.
+   * <p>
+   * When the merge is completed, or fails, the {@link MergeOfflineRegionsCallback} will be invoked on the main thread.
+   * <p>
+   * The secondary database may need to be upgraded to the latest schema.
+   * This is done in-place and requires write-access to the provided path.
```

Comment:
Is it necessary to pass this burden to the developer instead of copying the file to a temporary location as part of the SDK bindings?

Reply:
Initially, I was worried about unnecessarily increasing the merge time for databases that are already in a writable directory. However, we can avoid that with `File#canWrite` check and copy only if needed.